src/build_log.cc

```diff
@@ -393,6 +398,18 @@ bool BuildLog::Recompact(const string& path, const BuildLogUser& user,
   for (size_t i = 0; i < dead_outputs.size(); ++i)
     entries_.erase(dead_outputs[i]);
 
+  //secure owning uid and gid
+  if (access(path.c_str(), R_OK) < 0) {
```

Comment:
Why is this call necessary? Why not just use `stat` and set `found_uid_gid` based on its return code?

Reply:
Hello!

This call is not directly nesessary, but i wanted it to keep it out of stat for one reason.
It is more code to check after a failed access(...) if it is due to a missing file / directory, or some internal error that we should report. So i just stayed with a seperated access, which would indicate that we cannot read the file in any case and nothing with internal stats went wrong.

Anyway, i dont have a _that_ strong opinion on that, we can also go the path of "if stat failed just stay with the uid / gid from now"

What would you like to see?

Reply:
I'd rather we just call `stat` rather than trying to ask if we're OK. "Ask for forgiveness rather than permission" kind of thing.

Reply:
Done! :)